packages/editor/src/utils/firebaseUtils.tsx

```diff
@@ -0,0 +1,163 @@
+import { initializeApp } from "firebase/app";
+import {
+  GithubAuthProvider,
+  getAuth,
+  signInWithPopup,
+  signOut,
+} from "firebase/auth";
+import {
+  collection,
+  doc,
+  getDoc,
+  getDocs,
+  getFirestore,
+} from "firebase/firestore";
+import toast from "react-hot-toast";
+import { useRecoilCallback } from "recoil";
+import { SavedWorkspaces, Workspace, settingsState } from "../state/atoms.js";
+
+const firebaseConfig = {
+  apiKey: "AIzaSyA_WZrM0sWpOt3oKmPp_D77rS7TsqaTC-w",
+  authDomain: "penrose-test.firebaseapp.com",
+  projectId: "penrose-test",
+  storageBucket: "penrose-test.appspot.com",
+  messagingSenderId: "997650567502",
+  appId: "1:997650567502:web:f46626596ca78b065d8e1f",
+  measurementId: "G-GMD2T1MJ24",
+};
+
+const firebaseApp = initializeApp(firebaseConfig);
+export const authObject = getAuth(firebaseApp);
+export const db = getFirestore(firebaseApp);
+export const oauthProvider = new GithubAuthProvider();
+oauthProvider.addScope("gist");
+
+// Auth utils
+export const logInWrapper = () =>
+  useRecoilCallback(({ set }) => async () => {
+    signInWithPopup(authObject, oauthProvider)
+      .then((result) => {
+        const credential = GithubAuthProvider.credentialFromResult(result);
+        if (credential !== null) {
+          /**
+           * Save access token to local state
+           * settingsEffect propogates to local storage
+           */
+          set(settingsState, (prevState) => ({
+            ...prevState,
+            githubAccessToken:
+              credential.accessToken != undefined // For type safety, assign null if undefined
+                ? credential.accessToken
+                : null,
+          }));
+
+          toast.success(`Logged in as ${result.user.displayName}`);
+        }
+      })
+      .catch((error) => {
+        toast.error("Error logging in");
```

Comment:
```suggestion
        toast.error(`Error logging in: ${error}`);
```
if the error messages are meaningful, it'd still be good to propagate them to the user for reporting.